app/models/group_offer.rb

```diff
@@ -25,6 +25,35 @@ class GroupOffer < ApplicationRecord
 
   scope :in_department, (-> { where.not(department_id: nil) })
 
+  scope :active_group_assignments_between, lambda { |start_date, end_date|
+    joins(:group_assignments).merge(GroupAssignment.active_between(start_date, end_date))
+  }
+
+  def has_active_assignments_between?(start_date, end_date)
```

Comment:
Maybe rename that to `has_active_group_assignments_between?`

Reply:
For the sake of explicity yes, but don't blame me for rubocop hassling for too long lines, if only the method name without parameters fills 50% of the allowed line lengths 😉